src/overlay/OverlayManagerTests.cpp

```diff
@@ -118,24 +118,89 @@ class OverlayManagerTests
     {
         auto cfg = getTestConfig();
         cfg.TARGET_PEER_CONNECTIONS = 5;
+        cfg.KNOWN_PEERS = threePeers;
+        cfg.PREFERRED_PEERS = fourPeers;
         app = createTestApplication<ApplicationStub>(clock, cfg);
     }
 
     void
-    test_addPeerList()
+    test_addPeerList(bool async = false)
```

Comment:
as you're touching this, can you rename the `test_*` functions to use something closer to our naming convention?